### PR TITLE
VPN: Allow IPv6 local network exclusions

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1207603085593419/task/1211962786867930?focus=true

### Description

Exclude local networks is now honored for IPv6 routing

### Testing Steps

1. Start the VPN.
2. Enable "exclude local networks"
3. Try connecting to / pinging a device in your LAN through IPv6.  It should succeed.
4. Check "what's my ip address" using the browser and ensure it's the VPN IP for both IPv4 and 6.
5. Disable "exclude local networks"
6. Try connecting to / pinging a device in your LAN through IPv6.  It should fail.
7. Check "what's my ip address" using the browser and ensure it's the VPN IP for both IPv4 and 6.

You can also run `netstat -nr -f inet6` and ensure the routing makes sense, although this might be confusing to check.

### Impact and Risks

High.  We're touching routing tables for the VPN.

#### What could go wrong?

We could break routing, although this was done in a way that should minimize risk, as we're still including all-IPv6 into the VPN.

### Quality Considerations

NA

### Notes to Reviewer

Please note most of the lines changed are tests, the logic changes are quite focused.

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> Adds IPv6 routing (::/0) with system exclusions and conditional ULA handling, updates resolver logic, and refactors tests accordingly.
> 
> - **Routing ranges**:
>   - Add `fullIPv6RoutingRange` (`::/0`) and `localIPv6NetworkRange` (`fc00::/7`).
>   - Refine `alwaysExcludedIPv6Range` (loopback, link-local, multicast) and comments.
>   - Rename `publicNetworkRange` to `publicIPv4NetworkRange`.
> - **Resolver (`VPNRoutingTableResolver`)**:
>   - Include IPv6 defaults in included routes and IPv6 system ranges in excluded routes.
>   - Conditionally add/remove IPv6 ULA with `excludeLocalNetworks`.
>   - Generate DNS host routes with correct prefixes (`/32` IPv4, `/128` IPv6).
> - **Tests**:
>   - Update integration/unit tests to validate IPv6 behavior: `::/0` inclusion, system exclusions, and ULA handling.
>   - Adjust assertions to use `publicIPv4NetworkRange` and new IPv6 ranges; add IPv6 DNS route prefix checks.
>   - Remove/trim some non-essential/perf/conflict tests and revise overlap checks to IPv4-only where appropriate.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 73d1705f4affad46bf667d71932a7d030c2c2ff4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingRange.swift [hunk 1]
```diff
@@ -28,24 +28,27 @@ public enum VPNRoutingRange {
     ]
 
     public static let alwaysExcludedIPv6Range: [VPN.IPAddressRange] = [
-        "fe80::/10",  /* link local */
-        "ff00::/8",   /* multicast */
-        "fc00::/7",   /* local unicast */
```
Comment (diegoreymendez) [inline SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingRange.swift:33]: This was moved to local networks, and is enabled / disabled through the toggle.

FWIW, this didn't have any effect before this PR as we were NOT using the exclusion list in IPv6 at all.

File: SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingRange.swift [hunk 2]
```diff
@@ -28,24 +28,27 @@ public enum VPNRoutingRange {
...
+        "ff00::/8",   /* ff00:: - ffff::      Multicast */
     ]
 
     public static let localNetworkRangeWithoutDNS: [VPN.IPAddressRange] = [
         "172.16.0.0/12",  /* 255.240.0.0 */
         "192.168.0.0/16", /* 255.255.0.0 */
     ]
 
+    public static let localIPv6NetworkRange: [VPN.IPAddressRange] = [
+        "fc00::/7",   /* fc00:: - fdff::      Unique Local Addresses (ULA) - IPv6 private networks */
```
Comment (diegoreymendez) [inline SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingRange.swift:42]: Private range, please double check.  This is what'll be excluded.

File: SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingRange.swift [hunk 3]
```diff
@@ -100,6 +103,10 @@ public enum VPNRoutingRange {
         "208.0.0.0/4",      /* 208.0.0.0 - 223.255.255.255 */
         /* 224.0.0.0            224.0.0.0/4 - 239.255.255.255   Multicast */
         /* 240.0.0.0            240.0.0.0/4 - 255.255.255.255   Class E */
-        "::/0" /* IPv6 */
     ]
+
+    /// The full IPv6 routing range (all addresses).
+    /// Unlike IPv4 which uses explicitly carved ranges, IPv6 includes all addresses
+    /// and relies on excluded routes taking precedence per Apple's NEIPv6Settings documentation.
+    public static let fullIPv6RoutingRange: VPN.IPAddressRange = "::/0"
```
Comment (diegoreymendez) [inline SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingRange.swift:111]: Just making this more semantically meaningful.  To make it less likely we'll see trouble, I'll rely on Apple claiming exclusions take precedence over inclusions.

So I'm keeping this in included routes, so the VPN claims everything for IPv6, and we'll only specify what to exclude for it.

File: SharedPackages/VPN/Tests/VPNTests/Routing/VPNRoutingIntegrationTests.swift [hunk 4]
```diff
@@ -130,33 +130,6 @@ final class VPNRoutingIntegrationTests: XCTestCase {
 
     // MARK: - Edge Case and Boundary Tests
 
-    /// Verifies that VPN routing remains fast and responsive even with unusually complex DNS configurations
-    func testComplexDNSConfigurationsRemainPerformant() {
```
Comment (diegoreymendez) [inline SharedPackages/VPN/Tests/VPNTests/Routing/VPNRoutingIntegrationTests.swift:134]: Lots of testing changes.  I focused on value - removing tests that were adding no value, updating existing tests to be more atomic and provide clear value, and adding new tests to cover areas we weren't properly covering.

That said, don't assume the tests prove that the ranges we chose are correct - I still suggest carefully reviewing the ranges changed in this PR attentively.

File: SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingTableResolver.swift [hunk 5]
```diff
@@ -40,20 +40,22 @@ struct VPNRoutingTableResolver {
...
             routes += VPNRoutingRange.localNetworkRangeWithoutDNS
+            routes += VPNRoutingRange.localIPv6NetworkRange
         }
 
         return routes
     }
 
     var includedRoutes: [IPAddressRange] {
-        var routes = VPNRoutingRange.publicNetworkRange + dnsRoutes()
+        var routes = VPNRoutingRange.publicIPv4NetworkRange + [VPNRoutingRange.fullIPv6RoutingRange] + dnsRoutes()
```
Comment (diegoreymendez) [inline SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingTableResolver.swift:54]: This just makes sure the IPv6 range we took out of publicNetworkRange is still included.

File: SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingTableResolver.swift [hunk 6]
```diff
@@ -40,20 +40,22 @@ struct VPNRoutingTableResolver {
     }
 
     var excludedRoutes: [IPAddressRange] {
-        var routes = VPNRoutingRange.alwaysExcludedIPv4Range
+        var routes = VPNRoutingRange.alwaysExcludedIPv4Range + VPNRoutingRange.alwaysExcludedIPv6Range
```
Comment (diegoreymendez) [inline SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingTableResolver.swift:43]: This is a change in behavior, please review carefully.

File: SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingRange.swift (hunk 2 above)
Comment (samsymons) [reply SharedPackages/VPN/Sources/VPN/Routing/VPNRoutingRange.swift:42]: Looks good per https://www.rfc-editor.org/rfc/rfc4193.
